### ISSUE TITLE
dnsmasq: cached dnssec responses fail validation

### ISSUE THREAD

**Issue by mweinelt** [freifunk-gluon/gluon#1362]
I tried using the dnsmasq cache on the nextnode address today and my unbound, who is reconfigured to use that cache, fails to validate DNSSEC secured responses, that are pulled from the cache, and therefore makes some domains unusable. 

Uncached responses otoh work just fine.

**Comment by christf** [comment]
dnsmasq seems to not store the authenticity information in its cache by default. Could you try using --proxy-dnssec in the config file for dnsmasq?

**Comment by mweinelt** [comment]
While the option is shown in the `--help` section DNSSEC support is only built-in in `dnsmasq-full`.

**Comment by christf** [comment]
--dnssec is not needed for --proxy-dnssec because on the latter no validation happens.

you can run
 ps www|grep dnsmasq
to find the instance that is run by the dnsmasq user. copy the command line, kill this very instance and run this dnsmasq command again with --proxy-dnssec option added on the cli. Then, please verify if dnssec caching works.

**Comment by NeoRaider** [comment]
"Fixed" in 543eb178824e8ed8a6f385fe1ebdf0d7ca709be3.

**Comment by mweinelt** [comment]
Just because I was curious what it would take:
`dnsmasq-full` plus dependencies comes down to roughly 784 kB of additional disk space.

```
136.0K dnsmasq-full_2.80-1_mipsel_24kc.ipk
120.0K kmod-ipt-ipset_4.14.78-1_mipsel_24kc.ipk
16.0K kmod-nf-conntrack-netlink_4.14.78-1_mipsel_24kc.ipk
212.0K libgmp_6.1.2-1_mipsel_24kc.ipk
8.0K libmnl_1.0.4-1_mipsel_24kc.ipk
36.0K libnetfilter-conntrack_2017-07-25-e8704326-1_mipsel_24kc.ipk
244.0K libnettle_3.4-1_mipsel_24kc.ipk
12.0K libnfnetlink_1.0.1-1_mipsel_24kc.ipk
784.0K total
```
